Gems/Streamer/StreamerProfiler/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+
+# Add the Streamer.Static target
+ly_add_target(
+    NAME StreamerProfiler.Static STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        streamerprofiler_files.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            Code
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzCore
+            Gem::ImGui.imguilib
+)
+
+# Here add Streamer target, it depends on the Streamer.Static
+ly_add_target(
+    NAME StreamerProfiler ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
+    NAMESPACE Gem
+    FILES_CMAKE
+        streamerprofiler_shared_files.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            Code
+    BUILD_DEPENDENCIES
+        PRIVATE
+            Gem::StreamerProfiler.Static
+)
+
+# By default, we will specify that the above target Streamer would be used by
+# Client and Server type targets when this gem is enabled.  If you don't want it
+# active in Clients or Servers by default, delete one of both of the following lines:
+ly_create_alias(NAME StreamerProfiler.Clients NAMESPACE Gem TARGETS Gem::StreamerProfiler)
+ly_create_alias(NAME StreamerProfiler.Servers NAMESPACE Gem TARGETS Gem::StreamerProfiler)
```

Comment:
Since the name of the gem is "Streamer", you will need to create an alias for GemVariants that is fits the format of "Streamer.<variant>". 

This is to allow interoperability with the enable gem functionality which by convention attempts to add gem modules as a dependencies of the project launcher by looking for [${GemName}.<Variants>](https://github.com/o3de/o3de/blob/development/cmake/Gems.cmake#L243-L256) target.
i.e
```
ly_create_alias(NAME Streamer.Clients NAMESPACE Gem TARGETS Gem::Streamer)
```

O

Reply:
The intention of the gem is to house all parts related to AZ::IO::Streamer's implementation (the interface will remain in AzCore). Features like the profiler are optional parts of the gem. Last time we talked, the gem system doesn't support this quite yet so we came up with the above work-around. I wanted to get this gem in now in preparation of moving the AZ::IO::Streamer implementation there in the future so future PRs will be a little smaller, hence there's no Streamer target yet. Another reason to get this in as-is is that it puts a highlight on needing the feature to select different parts of a gem rather only allowing a single module in order to start consolidating the fine-grained gems O3DE has and allow companies to package multiple products or product features in a single gem.
Please advise if any plans around this have changed.

Reply:
Well the portion of the Gem System that doesn't support the optional features is the ability for CMake to associate what targets to activate for a gem based solely on its gem name.

Currently the convention is given a gem name specified in the call to [ly_enable_gem](https://github.com/o3de/o3de/blob/development/cmake/Gems.cmake#L142-L153) we check if there is a CMake target that matches the pattern of ${GemName}.${Variant} for all Gem Variants that exist in the engine as created by [ly_set_gem_variant_to_load](https://github.com/o3de/o3de/blob/development/cmake/Gems.cmake#L101-L105).

The CMake logic to determine the list of gems to load visits all the gem dependencies of a target associated with a gem variant and look for any `<dependent gem name>.<variant name>` target to determine if we should add that MODULE target to the list of generated `cmake_dependencies.*.setreg` file.

Why not make the "StreamerProfiler" a sub gem that resides within the "Streamer" gem folder?
All that would require is an additional gem.json file within a subfolder of the "Streamer" gem and an "external_subdirectories" entry for the "StreamerProfiler" directory?

With that the Project Manager and the o3de python scripts would recognize the "StreamerProfiler" feature as a gem.